### ISSUE TITLE
Faster Inference Speed

### ISSUE THREAD

**Issue by yt2639** [prs-eth/Marigold-DC#6]
Hii authors, thank you for your great work and for releasing the code. I am wondering if Marigold-DC also supports faster inference using LCM (https://huggingface.co/prs-eth/marigold-depth-lcm-v1-0) as what the original Marigold does?

Right now on A40 (48GB GPU Memory)  machine, inference on single image takes around 1.5 minutes. I am looking for some ways to improve the speed.

Thanks! 😄

**Comment by otenim** [comment]
@yt2639 

hi

try this

```python3
# Choose ckpt
ckpt_path = "prs-eth/marigold-v1-0"
# for lcm model: ckpt_path = "prs-eth/marigold-lcm-v1-0"

# Try low precision inference
pipe = MarigoldDepthCompletionPipeline.from_pretrained(
  ckpt_path,
  prediction_type: "depth",
  torch_dtype: torch.bf16  # NOTE: use bf16 instead of fp16; fp16 produced nan values for my case
).to("cuda")

# Use lightweight vae
pipe.vae = AutoencoderTiny.from_pretrained(
  "madebyollin/taesd"
   torch_dtype=torch.bf16,
).to("cuda")
```
hope this work well

**Comment by otenim** [comment]
@yt2639 

another tips here.

Try SGD optimizer instead of the original Adam with doubled learning rate of latent variable like;

```python3
optimizer = torch.optim.SGD([
  {"params": [scale, shift], "lr": 0.005},
  {"params": [pred_latent], "lr": 0.1}, # <- original: 0.05
])
```
This worked really well in my case. Got good depth error with ~10 denoising steps.

**Comment by massimilianoviola** [comment]
That is correct, there are several ways to speed up inference:
- `bfloat16` enables 2× faster inference while retaining sufficient gradient precision for backpropagation (`float16` does not handle small gradients and clips them to zero)
- `TinyVAE` adds another 2× with only a minor loss of accuracy.
- Reducing the number of steps is also a possibility, especially indoors.
- Model compilation is also supported and can enabled via:
  ```python
  pipe.vae = torch.compile(pipe.vae, mode="reduce-overhead", fullgraph=True)
  pipe.unet = torch.compile(pipe.unet, mode="reduce-overhead", fullgraph=True)```
Hope this helps!
